### PR TITLE
[4.x] exclude bake commands

### PR DESCRIPTION
**Pull request by saeideng**
Closes: #433

exclude bake commands when bake plugin is not installed

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -33,4 +39,112 @@ class Plugin extends BasePlugin
...
+    /**
+     * Add migrations commands.
+     *
+     * @param \Cake\Console\CommandCollection $collection The command collection to update
+     * @return \Cake\Console\CommandCollection
+     */
+    public function console(CommandCollection $collection): CommandCollection
+    {
+        if (class_exists('Bake\Command\SimpleBakeCommand')) {
+            $commands = $collection->discoverPlugin($this->getName());
```
Comment (markstory) [inline src/Plugin.php:52]: Would it be possible to always use `discoverPlugin()` and when bake is not present unset the commands that depend on Bake? That might let you avoid duplicating code from the command scanner.

Comment (saeideng) [reply src/Plugin.php:52]: unfortunately using `discoverPlugin` causes the error

I explained here:
https://github.com/cakephp/migrations/issues/433#issuecomment-593250034

Comment (saeideng) [reply src/Plugin.php:52]: I can add them manually
what do you think?

Comment (markstory) [reply src/Plugin.php:52]: Adding commands manually is how I solved this in authorization as well.

Comment (saeideng) [reply src/Plugin.php:60]: done.

File: templates/bake/config/snapshot.twig [hunk 2]
```diff
@@ -40,12 +40,12 @@ class {{ name }} extends AbstractMigration
      */
     public function up()
     {
-{% element 'Migrations.create-tables' {
+{{~ element('Migrations.create-tables', {
```
Comment (markstory) [inline templates/bake/config/snapshot.twig:43]: Should this be `{{-`?

Comment (saeideng) [reply templates/bake/config/snapshot.twig:43]: no , then it removes spaces + new lines from the beginning
see the result after replacing with `{{-`

```php
    public function up()
    {        $this->table('posts')
            ->addColumn('title', 'string', [
                'default' => null,
                'limit' => 1000,
                'null' => false,
            ])
```

Comment (saeideng) [reply templates/bake/config/snapshot.twig:43]: in my test in local , current code works well

**Pull request merged**